crates/sargon/src/signing/signatures_outecome_types/maybe_signed_transactions.rs

```diff
@@ -1,35 +1,32 @@
 use crate::prelude::*;
 
 #[derive(Debug, PartialEq, Eq, Clone)]
-pub(crate) struct MaybeSignedTransactions {
+pub(crate) struct MaybeSignedTransactions<S: Signable> {
```

Comment:
Looks like this can be made generic over `T: SignableID` only, not the whole `Signable` trait? Given my other review comments?